apps/api/src/app/step-schemas/usecases/get-step-schema/get-step-schema.usecase.ts

```diff
@@ -1,39 +1,149 @@
-import { Injectable } from '@nestjs/common';
-import {
-  ActionStepEnum,
-  actionStepSchemas,
-  ChannelStepEnum,
-  channelStepSchemas,
-  Schema,
-  StepType,
-} from '@novu/framework';
+import { BadRequestException, Injectable } from '@nestjs/common';
+
+import { JsonSchema, Schema, StepType } from '@novu/framework';
+import { NotificationStepEntity, NotificationTemplateRepository } from '@novu/dal';
 
-import { GetStepSchemaCommand } from './get-step-schema.command';
-import { SchemaOutput } from '../../types';
+import {
+  GetStepTypeSchemaCommand,
+  GetExistingStepSchemaCommand,
+  GetStepSchemaCommand,
+} from './get-step-schema.command';
+import { StepSchemaDto } from '../../dtos/step-schema.dto';
+import { mapStepTypeToOutput, mapStepTypeToResult } from '../../types';
 
 @Injectable()
 export class GetStepSchema {
-  async execute(command: GetStepSchemaCommand): Promise<SchemaOutput> {
-    return StepTypeToOutputMap[command.stepType];
+  constructor(private readonly notificationTemplateRepository: NotificationTemplateRepository) {}
+
+  async execute(command: GetStepSchemaCommand): Promise<StepSchemaDto> {
+    if (isGetByStepType(command)) {
+      return { controls: buildControlsSchema(command.stepType), variables: buildVariablesSchema() };
+    }
+
+    if (isGetByExistingStep(command)) {
+      const { currentStep, previousSteps } = await this.findSteps(command);
+
+      return {
+        controls: buildControlsSchema(currentStep?.template?.type as StepType),
+        variables: buildVariablesSchema(previousSteps),
+      };
+    }
+
+    throw new BadRequestException('Invalid command');
```

Comment:
Could we create a new error type that includes all relevant information and issues? The current error message doesn’t provide enough detail for the API user to effectively resolve their problems.